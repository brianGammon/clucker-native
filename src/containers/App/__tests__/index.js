import React from 'react';
import { shallow } from 'enzyme';
import firebase from 'react-native-firebase';
import { App } from '..';

jest.mock('../../../navigation/RootNavigator', () => 'Test');

describe('App component shallow', () => {
  test('Not logged in', () => {
    const onAuthStateChangedMock = jest.fn(cb => cb(null));
    firebase.auth = jest.fn(() => ({
      onAuthStateChanged: onAuthStateChangedMock,
    }));
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Logged in', () => {
    const listenToUserSettingsMock = jest.fn();
    const onAuthStateChangedMock = jest.fn(cb => cb({ uid: 'Test123' }));
    firebase.auth = jest.fn(() => ({
      onAuthStateChanged: onAuthStateChangedMock,
    }));
    const wrapper = shallow(<App listenToUserSettings={listenToUserSettingsMock} />);
    expect(wrapper).toMatchSnapshot();
    expect(listenToUserSettingsMock).toHaveBeenCalledWith('Test123');
  });
});
