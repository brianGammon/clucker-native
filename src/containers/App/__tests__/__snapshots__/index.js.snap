// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component shallow Logged in 1`] = `<Test />`;

exports[`App component shallow Not initialized 1`] = `<_default />`;

exports[`App component shallow Not logged in 1`] = `<Test />`;
